cli/command/task.go

```diff
@@ -10,6 +16,221 @@ func NewTaskCommand(b *CommandBase) *TaskCommand {
+		Name:  "task",
+		Usage: "manage layer0 tasks",
+		Subcommands: []cli.Command{
+			{
+				Name:      "create",
+				Usage:     "create a new task",
+				Action:    t.create,
+				ArgsUsage: "ENVIRONMENT NAME DEPLOY",
+				Flags: []cli.Flag{
+					cli.IntFlag{
+						Name:  "copies",
+						Value: 1,
+						Usage: "number of copies of deploy to run (default: 1)",
+					},
+					cli.StringSliceFlag{
+						Name:  "env",
+						Usage: "environment variable override in format 'CONTAINER:VAR=VAL' (can be specified multiple times)",
+					},
+				},
+			},
+			{
+				Name:      "delete",
+				Usage:     "delete a Task",
+				ArgsUsage: "NAME",
+				Action:    t.delete,
+			},
+			{
+				Name:      "list",
+				Usage:     "list all Tasks",
+				Action:    t.list,
+				ArgsUsage: " ",
+			},
+			{
+				Name:      "read",
+				Usage:     "describe a Task",
+				Action:    t.read,
+				ArgsUsage: "NAME",
+			},
+			{
+				Name:      "logs",
+				Usage:     "get the logs for a task",
+				Action:    t.logs,
+				ArgsUsage: "NAME",
+				Flags: []cli.Flag{
+					cli.IntFlag{
+						Name:  "tail",
+						Usage: "number of lines from the end to return",
+					},
+					cli.StringFlag{
+						Name:  "start",
+						Usage: "the start of the time range to fetch logs (format: YYYY-MM-DD HH:MM)",
+					},
+					cli.StringFlag{
+						Name:  "end",
+						Usage: "the end of the time range to fetch logs (format: YYYY-MM-DD HH:MM)",
+					},
+				},
+			},
+		},
+	}
+}
+
+func (t *TaskCommand) create(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "ENVIRONMENT", "DEPLOY", "NAME")
+	if err != nil {
+		return err
+	}
+
+	overrides, err := parseOverrides(c.StringSlice("env"))
+	if err != nil {
+		return err
+	}
+
+	environmentID, err := t.resolveSingleEntityIDHelper("environment", args["ENVIRONMENT"])
+	if err != nil {
+		return err
+	}
+
+	deployID, err := t.resolveSingleEntityIDHelper("deploy", args["DEPLOY"])
+	if err != nil {
+		return err
+	}
+
+	req := models.CreateTaskRequest{
+		TaskName:           args["NAME"],
+		EnvironmentID:      environmentID,
+		DeployID:           deployID,
+		ContainerOverrides: overrides,
+	}
+
+	jobID, err := t.client.CreateTask(req)
+	if err != nil {
+		return err
+	}
+
+	return t.waitOnJobHelper(c, jobID, "creating", func(taskID string) error {
+		task, err := t.client.ReadTask(taskID)
+		if err != nil {
+			return err
+		}
+
+		return t.printer.PrintTasks(task)
+	})
+}
+
+func (t *TaskCommand) delete(c *cli.Context) error {
+	return t.deleteHelper(c, "task", func(taskID string) (string, error) {
+		return t.client.DeleteTask(taskID)
+	})
+}
+
+func (t *TaskCommand) list(c *cli.Context) error {
+	taskSummaries, err := t.client.ListTasks()
+	if err != nil {
+		return err
+	}
+
+	return t.printer.PrintTaskSummaries(taskSummaries...)
+}
+
+func (t *TaskCommand) read(c *cli.Context) error {
+
+	taskSummaries, err := t.client.ListTasks()
+	if err != nil {
+		return err
+	}
+
+	taskExists := func(id string) bool {
```

Comment:
it would be faster to have `taskExists` be a `map[string]bool`, e.g.:
```
taskExists := map[string]bool{}
for _, taskSummary := range taskSummaries {
    taskExists[taskSummary.TaskID] = true
}

...
for _, taskID := range taskIDs {
    if taskExists[taskID] {
...
```

Reply:
I was wondering about this but wasn't sure if the conversion to map would be okay. Thanks though I agree.